### ISSUE TITLE
Oldest Et futurum Requiem has a problem to make a crash

### ISSUE THREAD

**Issue by SatakeKanrin** [Roadhog360/Et-Futurum-Requiem#143]
When running the mod with ImmersiveEngineering,Mariculture and thaumic bases,HMLC reports exceptions among ETR,IE,Mariculture and TB.

It said that:"Fatal errors were detected during the transition from PREINITIALIZATION to INITIALIZATION. Loading cannot continue"

I sincerely hope you can fix this crash bug among these mods.Also I admire your perseverance to complete the migration from high version to low version.
Mod Version 2.4.0

**Comment by Roadhog360** [comment]
Yes, when making issues please use the template. I ask for crucial information in that template, yet for whatever reason, A lot of people simply skip it, even though the option to use the template is way bigger.

@GregoriusT do you know how to force people to use the template? This happens a bit too much and I'd like to prevent it.

**Comment by GregoriusT** [comment]
The real "issue" there is, the "open blank Issue" thing always exists. And then they dont see any of the things you ask for.

Also the ONE useful thing the template could ask for is NOT on your template. How are people supposed to give you Logs if you dont tell them that Logs exist in the first place, and where to find them too. XD

**Comment by SatakeKanrin** [comment]
Actually,I put these mods together instead they didn't cause the crash.But the vision after entering the world was a kind of pure color.
Here is the log
[Uploading latest.log…]()

**Comment by GregoriusT** [comment]
Uhm your upload got corrupted. Try again please. I have no Idea how GitHub fucked up this one...

**Comment by OvermindDL1** [comment]
Github didn't mess anything up, they just didn't wait for the upload to complete before sending the message.

**Comment by SatakeKanrin** [comment]
[latest.log](https://github.com/Roadhog360/Et-Futurum-Requiem/files/9076005/latest.log)

**Comment by GregoriusT** [comment]
Ah thank you, here is the relevant bit!

```
java.lang.RuntimeException: There is a potion ID Conflict: class blusunrize.immersiveengineering.common.util.IEPotions$IEPotion and class ganymedes01.etfuturum.potion.PotionLevitation, potion id:27
	at net.tclproject.mysteriumlib.asm.fixes.MysteriumPatchesFixesEnchantments.Potion(MysteriumPatchesFixesEnchantments.java:59)
	at net.minecraft.potion.Potion.<init>(Potion.java:66)
	at blusunrize.immersiveengineering.common.util.IEPotions$IEPotion.<init>(IEPotions.java:95)
	at blusunrize.immersiveengineering.common.util.IEPotions.init(IEPotions.java:35)
	at blusunrize.immersiveengineering.common.IEContent.init(IEContent.java:508)
	at blusunrize.immersiveengineering.ImmersiveEngineering.init(ImmersiveEngineering.java:99)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```

You got a Potion ID Conflict! That should be fixable using the Configs. ^^

**Comment by SatakeKanrin** [comment]
I'm deeply grateful!

**Comment by SatakeKanrin** [comment]
Anti ID Confilct didn't report it so I had not treated the crash as an ID confilct before.

**Comment by GregoriusT** [comment]
Yeah everyone has their own ID system for Potion Conflicts, some just overwrite Potions, others crash when ID Conflicts happen, and everyone has a different Config for it. Quite annoying indeed.

Anyways I guess I can close this now that you solved it yourself. ^^

**Comment by Roadhog360** [comment]
Please upload the new crash report. This one line does not tell us what mod is causing it.

**Comment by SatakeKanrin** [comment]
[latest.log](https://github.com/Roadhog360/Et-Futurum-Requiem/files/9077419/latest.log)
![image](https://user-images.githubusercontent.com/108935011/178112003-f60c7875-9605-4af9-acf5-c913a1693e15.png)

**Comment by GregoriusT** [comment]
@Roadhog360 your config needs to say 0 - 255 not 0 - 256, also i am not sure if it is actually limited to positive integers or not, so it might only be 0 - 127.

**Comment by Roadhog360** [comment]
@SatakeKanrin `java.lang.IllegalArgumentException: Duplicate enchantment id! class mariculture.magic.enchantments.EnchantmentSpider and class ganymedes01.etfuturum.enchantment.SwiftSneak Enchantment ID:38`

Your crash is caused by two errors, one is this, you forgot to set Swift Sneak to a different ID.

Secondly, the out of bounds error is not caused by me, take a look:
`[17:34:40] [Client thread/ERROR]: Caught exception from **thaumicbases**
java.lang.ArrayIndexOutOfBoundsException: 256`
This error is caused by "thaumicbases".

@GregoriusT I'll set the max value of all potion IDs to Byte.MAX_VALUE or 127, levitation currently complies with that limit although you cannot see it because it is off-screen. Levitation complies with this. However in investigating your suggestion I have actually found that the number I chose for enchantment IDs, 256, is completely arbitrary. I guess I was following in the steps of ganymedes01 but enchantments are actually a short so that should be Short.MAX_VALUE or 32767, not arbitrarily limited to 256. I'll correct this error in 2.4.1.
